### PR TITLE
Custom excel report

### LINKED ISSUE #29: `affirm_report_excel` formatting
1. Add label to row 1.
2. Start data printout on row 2; use croquet::add_labelled_sheet() similar-ish style for data printout format.
3. Add first sheet with summary of affirm report.

### LINKED ISSUE #34: affirmation_name checks
Allow `affirmation_name` check to be more flexible to allow patterns like:

`affirmation_name = "{data_frames}{str_pad(id, width = 3, pad = '0')}"`

### PR DESCRIPTION
**Pull request by shannonpileggi**
Closes #29 
Closes #34

### CODE REVIEW

**Comment by shannonpileggi** [comment]
still wip

**Comment by shannonpileggi** [comment]
Notes on next to do steps:

- [ ] fix R cmd check warning/notes
- [ ] fix/remove documentation for dot functions in utils
- [ ] fix iteration over all affirmations
- [ ] merge cell for labels in 1st row of sheet?

**Comment by shannonpileggi** [comment]
Are there additional unit tests I should add?

Some assumptions I made about the excel report format that a user may want to customize - what do you think? Should I add in some flexibility in this PR, or in another issue?
1. In the first summary sheet, dropping NA columns that were not submitted as arguments to `affirm_`. 
2. In the first summary sheet, the column ordering.
3. In all sheets, min/max defaults for column widths.
4. In both sheets, the addition of a "Notes" column.

**Comment by shannonpileggi** [comment]
need to re-think checking on affirmation_name in excel report, doesn't allow something like this: `affirmation_name = "{data_frames}{str_pad(id, width = 3, pad = '0')}"`

**Comment by shannonpileggi** [comment]
> need to re-think checking on affirmation_name in excel report, doesn't allow something like this: `affirmation_name = "{data_frames}{str_pad(id, width = 3, pad = '0')}"`

Maybe this can be a separate issue to keep things simpler.

**Review by ddsjoberg — APPROVED** [review]
All looks good to me!

> Some assumptions I made about the excel report format that a user may want to customize - what do you think? Should I add in some flexibility in this PR, or in another issue?

I would add/discuss in another issue/PR

File: NAMESPACE [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
+export(.add_summary_sheet)
 export(.check_replace_argument)
+export(.compute_col_width)
 export(.data)
 export(.env)
+export(.identify_keep_data)
+export(.retrieve_labels)
 export(affirm_class)
 export(affirm_clean_join)
 export(affirm_close)
```
[the remark this answers is not in the document]
Comment (shannonpileggi) [reply NAMESPACE:14]: they don't need to be exported, let's discuss 😊

File: _pkgdown.yml [hunk 2]
```diff
@@ -27,6 +27,11 @@ reference:
 - contents:
   - affirm_init
   - affirm_report
+  - .add_affirmation_sheet
```
[the remark this answers is not in the document]
Comment (shannonpileggi) [reply _pkgdown.yml:30]: sounds good if we leave them as exported!

**Pull request merged**